community: add cuda team

ConnorBaker opened the pull request:
Adds an entry for the CUDA team on the NixOS community teams page.

ConnorBaker commented:
@NixOS/cuda-maintainers, @someoneserge, @samuela -- let me know if it's okay to add you as team members, and if I should make any corrections to the description or goals!

ConnorBaker commented:
@fricklerhandwerk is this similar to what you had in mind for the homepage addition?

fricklerhandwerk commented on community/teams/cuda.tt:
```diff
@@ -0,0 +1,61 @@
+    Discussions are held on Matrix, Discourse, and on GitHub issues and pull requests.
+  </p>
+  <p>
+    <a class="button" href="https://nixos.wiki/wiki/CUDA">Wiki</a>
```
```suggestion
```

Probably shouldn't link to the Wiki, it's all over the place in terms of quality. This page in particular seems not very insightful. One thing this team could do for starters would be updating the [Nixpkgs manual section on CUDA](https://nixos.org/manual/nixpkgs/unstable/#cuda) a bit, and find ways to make the Wiki page obsolete (so we can eventually delete it), e.g. by adding guides and tutorials to nix.dev.

SomeoneSerge replied:
Tbh the only reason I see the wiki page is still alive is that it links to cachix, which we don't include in the manual since there still isn't a reliable solution as to populating it, and no clear trust model. Replacing the wiki page with a clear cut tutorial on nix.dev sounds great

ConnorBaker replied:
Good points -- updated docs and tutorials would certainly be welcome and are something we need to work on.

fricklerhandwerk commented on community/teams/cuda.tt:
```diff
@@ -0,0 +1,61 @@
+  <h2>Goals</h2>
+  <ul>
+    <li>
+      support community efforts involving packaging or distributing CUDA-enabled software
+    </li>
+    <li>
+      document CUDA packaging best practices, guidelines, and patterns
+    </li>
+    <li>
+      improve the quality of CUDA packaging
+    </li>
+    <li>
+      make Nix/NixOS the first choice for users of CUDA-related software, especially in machine learning and scientific computing
```
Suggestion: order this by specificity. What you really care about and can work towards should be first. The last item is not really a goal but a recurrent task, but that is fine.

```suggestion
      make Nix and NixOS the first choice for users of CUDA-related software, especially in machine learning and scientific computing
    </li>
    <li>
      increase the number and improve the quality of CUDA packages
    </li>
    <li>
      document CUDA packaging best practices, guidelines, and patterns
    </li>
    <li>
      support community efforts involving packaging or distributing CUDA-enabled software
```

fricklerhandwerk replied:
One could change the list heading to "Goals and responsibilities", but as said, it's fine as it is. You can (and very probably will) refine it as you go.

ConnorBaker replied:
Fixed in newest

fricklerhandwerk commented on community/index.tt:
```diff
@@ -300,6 +300,10 @@
       <a class="button -primary" href="[%root%]community/teams/nix.html">Read more</a>
 
+      <h3>CUDA Team</h3>
+      <p>Responsible for all things CUDA-related in Nixpkgs.</p>
```
This is only my personal opinion, so feel free to ignore, since you're running that show: It's a bit too colloquial for my taste.

```suggestion
      <p>Maintains CUDA-related software in Nixpkgs.</p>
```

fricklerhandwerk commented on community/teams/cuda.tt:
```diff
@@ -0,0 +1,61 @@
+</div>
+
+<section class="governance-description">
+  The CUDA team is responsible for all things CUDA-related in <a href="https://nixpkgs.org">Nixpkgs</a>.
```
```suggestion
  The CUDA team is responsible for everything related to <a href="https://developer.nvidia.com/cuda-zone">CUDA</a> in <a href="https://github.com/NixOS/nixpkgs">Nixpkgs</a>.
```

We should not link to the same page we're on. Nixpkgs is primarily a GitHub repo. Also it probably makes sense to link to what CUDA is, as we can't expect everyone to know what that is. This page is for the general audience.

SomeoneSerge replied:
> Not a goal

Looking at https://nixos.org/community/teams/nix.html as an example, we could put this in "responsibilities"

ConnorBaker replied:
Fixed in newest

fricklerhandwerk reviewed (approved):
Looks good to me. Left a few superficial suggestions which you can ignore if you disagree.

SomeoneSerge commented:
>  let me know if it's okay to add you as team members

Sure

samuela commented:
> https://github.com/orgs/NixOS/teams/cuda-maintainers, @SomeoneSerge, @samuela -- let me know if it's okay to add you as team members, and if I should make any corrections to the description or goals!

yup, fine by me

ConnorBaker commented:
Ran into fun bug here: https://github.com/NixOS/nixos-homepage/issues/1049

Will do a gross workaround by manually adding entries to the `community.less` file to handle our team's addition.

ConnorBaker commented:
Thanks for the reviews! So far I've:

- Made the changes from reviews
- Added Serge and Samuel
- Added a Motivation section
- Changed Goals to Goals and Responsibilities
- Added a call for sponsors (@fricklerhandwerk is that appropriate for a teams page? If not, I can remove it!)
- Fixed an issue on the community page where the grid flipped out because we went past the hardcoded length (see #1049)
- Cracked my knuckles and wrote SVG by hand to bring us our own team icon!
- Filed a bug about the buttons at the bottom of the page touching when using semi-wide viewports (#1050)
  - I have a PR to kind-of fix it as well

Since these are substantial changes, I'm re-requesting review.

SomeoneSerge commented on community/teams/cuda.tt:
```diff
@@ -0,0 +1,127 @@
+  <h2>Goals and Responsibilities</h2>
+  <p>
+    The singular focus of the CUDA-maintainers team is to make Nixpkgs the best
+    resource for packages leveraging CUDA, especially in machine learning and
```
Nit: what does it mean to be "the best resource"? Books, manuals, and forums are also "resources", are comparing nixpkgs to a book? Maybe "deployment solution", so we limit the comparison to spack, docker, conda, pip, etc

ConnorBaker commented:
> Awesome! (I applaud the realistic emphasis on the heat dissipation...)

You should have seen the PCIe edge connector pinout I had drawn on it. I commented it out when I realized it'd be too small to be visible at the size show on the webpage!

fricklerhandwerk commented on community/index.tt:
```diff
@@ -288,6 +288,11 @@
 
+      [% PROCESS svg path="site-styles/assets/gfx-gpu.white.svg" %]
+      <h3>CUDA Team</h3>
+      <p>Making NixOS and Nixpkgs the first choice for users of CUDA-accelerated software.</p>
```
```suggestion
      <p>Making Nix and NixOS the first choice for users of CUDA-accelerated software.</p>
```

I suppose this is what we really want to say here. You cannot use Nixpkgs without Nix (or some clone).

ConnorBaker replied:
Fixed in newest

ConnorBaker commented:
@fricklerhandwerk what remains in terms of review / approval before merging?

SomeoneSerge commented:
@knedlsepp @cigrainger consider a follow-up PR extending the member list on the community page?

SomeoneSerge commented:
@knedlsepp @cigrainer please consider following up with your own PRs adding yourselves onto the page (also into the `lib.teams.cuda.members` in `nixpkgs`)!
